### ISSUE TITLE
Make package compatible with psr-18 and/or httplug 2.0

### ISSUE THREAD

**Issue by rpkamp** [rpkamp/mailhog-client#16]

**Comment by rpkamp** [comment]
Since the authors of php-http/httplug are saying that it will probably be superseded by psr-18 at some point it would make more sense to update to psr-18 instead. However at the moment there is only an adapter for Guzzle6 and some lesser know packages, and I don't want to force people into using Guzzle6 whereas they have a lot more options with php-http/httplug now.

I will be revisiting this once prs-18 gets some more implementations.

**Comment by brooke-heaton** [comment]
Yes. I honestly find this unusable without requiring php-http/httplug 2.0. I can't install this.

**Comment by brooke-heaton** [comment]
Yea, sorry this looks like an awesome and promising plugin for the Behat tests, but there's no way I can successfully get this installed with any php-http/client-implementation They all seem to require httplug 2.0.

I get this all day:

```
  Problem 1
    - Installation request for php-http/httplug (locked at v1.1.0, required as ^1.0) -> satisfiable by php-http/httplug[v1.1.0].
    - php-http/guzzle6-adapter 2.x-dev requires php-http/httplug ^2.0 -> satisfiable by php-http/httplug[2.0.x-dev].
    - php-http/guzzle6-adapter v2.0.0 requires php-http/httplug ^2.0 -> satisfiable by php-http/httplug[2.0.x-dev].
    - php-http/guzzle6-adapter v2.0.1 requires php-http/httplug ^2.0 -> satisfiable by php-http/httplug[2.0.x-dev].
    - Conclusion: don't install php-http/httplug 2.0.x-dev
    - Installation request for php-http/guzzle6-adapter ^2.0 -> satisfiable by php-http/guzzle6-adapter[2.x-dev, v2.0.0, v2.0.1].

```

**Comment by rpkamp** [comment]
I'm not sure whether I should upgrade to PSR-18 or httplug 2.0 (which is an extension of PSR-18).

Sounds like a nice thing to investigate for [hacktoberfest](https://hacktoberfest.digitalocean.com) :)

**Comment by rpkamp** [comment]
According to [this blog post](https://www.php-fig.org/blog/2018/07/the-http-client-psr/), httplug 1.0 and 2.0 should be fully compatible, so changing `^1.0` to `^1.0|^2.0` in `composer.json` should be enough. Trying that now.

**Comment by brooke-heaton** [comment]
Yea, I tried that. Total failure. I'm just using Mink at this point. Works well enough.

**Comment by rpkamp** [comment]
No I mean I need to update that in _this_ package, I did not mean you should update in your product. I can well believe that that won't work :slightly_smiling_face:

**Comment by rpkamp** [comment]
I've just released version 0.3.6 of this package, which supports HTTPlug 2.0: https://github.com/rpkamp/mailhog-client/releases/tag/0.3.6

Can you please try to install that version and let me know it works for you @brooke-heaton ?

**Comment by rpkamp** [comment]
Since there is no response I will just assume that it works :dancer:
